File: dfs_bfs/boj/p1260.py
# 각 노드의 인접 노드 배열로 담은 2차원 배열
import sys
from collections import deque

def dfs(graph, v, visited):
    visited[v] = True
    print(v,end=' ')

    for i in graph[v]:
        if not visited[i]:
            dfs(graph, i, visited)


def bfs(graph, start, visited):
    queue = deque([start])
    visited[start] = False

    while queue:
        v = queue.popleft()
        print(v,end=' ')

        for i in graph[v]:
            if visited[i]:
                queue.append(i)
                visited[i] = False

n, m, start = map(int,sys.stdin.readline().split())
graph = [[] for  i in range(n+1)]
visited = [False for i in range(n+1)]

for i in range(m):
        v1, v2 = map(int, sys.stdin.readline().split())
        graph[v1].append(v2)
        graph[v2].append(v1)

for i  in range(1, n+1):
    graph[i].sort()  #인접노드 작은 것부터 방문하기로 해서 정렬

dfs(graph, start, visited)
print()
bfs(graph, start, visited)

File: dfs_bfs/boj/P1260.java
package dfs_bfs.boj;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.LinkedList;
import java.util.Queue;
import java.util.StringTokenizer;

public class P1260 {
    static int n, m, v;
    static boolean[] visited;
    static int[][] graph;
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        StringTokenizer st = new StringTokenizer(br.readLine(), " ");
        n = Integer.parseInt(st.nextToken());
        m = Integer.parseInt(st.nextToken());
        v = Integer.parseInt(st.nextToken());

        graph = new int[n+1][n+1];

        for (int i = 0; i < m; i++) {
            st = new StringTokenizer(br.readLine(), " ");
            int x = Integer.parseInt(st.nextToken());
            int y = Integer.parseInt(st.nextToken());

            graph[x][y] = 1;
            graph[y][x] = 1;
        }

        visited = new boolean[n+1];

        DFS(v);
        System.out.println();

        Arrays.fill(visited, false);

        BFS(v);
        System.out.println();
    }

    static void DFS(int node) {
        if (visited[node]) {
            return;
        }
        
        visited[node] = true;
        System.out.print(node+" ");

        for (int i = 1; i <= n; i++) {
            if (graph[node][i] == 1 && visited[i] == false) {
                DFS(i);
            }
        }
    }

    static void BFS(int start) {
        Queue<Integer> que = new LinkedList<>();
        que.offer(start);
        visited[start] = true;

        while (!que.isEmpty()) {
            int node = que.poll();
            System.out.print(node+" ");

            for (int i = 1; i <= n; i++) {
                if (graph[node][i] == 1 && visited[i] == false) {
                    que.offer(i);
                    visited[i] = true;
                }
            }
        }

    }
    
}
